guilib/measure/uimeasurement.py

```diff
@@ -665,13 +683,14 @@ def __make_device(self, device_cls, device_name, **other_info):
         elif issubclass(device_cls, CameraABC):
             section = "camera_settings"
 
+        device_name = settings_info.more['name'] or settings_info.unique_name
         device.settings[section]['device_name'] = device_name
```

Comment:
Potential `NameError` if `section` is not defined (see conditions above). Perhaps add an `else` where you raise a `ValueError`?